.github/workflows/build-docker-image.yml

```diff
@@ -7,6 +7,10 @@ on:
   pull_request:
     branches: "*"
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
```

Comment:
Cancel in-progress builds when a new push occurs in a PR or on a branch. We should actually use that in other repos (magistrate?) too.
Azure builds in `vt` have `autoCancel` so that's fine:
```yml
pr:
  drafts: false
  autoCancel: true
```